src/snowflake/snowpark/_internal/ast/utils.py

```diff
@@ -1400,7 +1400,7 @@ def build_udtf(  # type: ignore[no-untyped-def] # TODO(SNOW-1491199) # Function
 
 
 # TODO(SNOW-1491199) - This method is not covered by tests until the end of phase 0. Drop the pragma when it is covered.
-def add_intermediate_stmt(ast_batch: AstBatch, o: Any) -> None:  # pragma: no cover
+def add_intermediate_stmt(o: Any) -> AstBuilder:  # pragma: no cover
```

Comment:
I was looking into this function earlier, I think it needs to be removed. Whichever objects are carrying `o._ast_stmt` now need to carry an `AstBuilder` instance and be added as a dependency wherever `add_intermediate_stmt` is used.

A more apt name for this might be `create_builder_from_stmt` if we still need it.